azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_events_provisioning.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from azure.iot.device.common.transport.pipeline_events_base import PipelineEvent
+
+
+class RegistrationResponseEvent(PipelineEvent):
+    """
+    A PipelineEvent object which represents an incoming RegistrationResponse event. This object is probably
+    created by some converter stage based on a pipeline-specific event
+    """
+
+    def __init__(self, request_id, status_code, key_values, response):
+        """
+        Initializer for RegistrationResponse objects.
+        :param request_id : The id of the request to which the response arrived.
+        :param status_code: The status code received in the topic.
+        :param key_values: A dictionary containing key mapped to a list of values that were extarcted from the topic.
+        :param response: The response received from a registration process
+        """
+        super(RegistrationResponseEvent, self).__init__()
+        self.rid = request_id
+        self.status_code = status_code
+        self.key_values = key_values
+        self.response = response
```

Comment:
Consider maybe using a model to represent a Registration Response that has these fields?

Reply:
renamed for now.

---
In reply to: [287546829](https://github.com/Azure/azure-iot-sdk-python-preview/pull/97#discussion_r287546829) [](ancestors = 287546829)